nixos/modules/services/blockchain/ethereum/geth.nix

```diff
@@ -0,0 +1,198 @@
+          Port number Go Ethereum will be listening on, both TCP and UDP.
+        '';
+      };
+
+      http = {
+        enable = lib.mkEnableOption "Go Ethereum HTTP API";
+        address = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            Listen address of Go Ethereum HTTP API.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8545;
+          description = ''
+            Port number of Go Ethereum HTTP API.
+          '';
+        };
+
+        apis = mkOption {
+          type = types.nullOr (types.listOf types.str);
+          default = null;
+          description = ''
+            APIs to enable over WebSocket
+          '';
+          example = ["net" "eth"];
+        };
+      };
+
+      websocket = {
+        enable = lib.mkEnableOption "Go Ethereum WebSocket API";
+        address = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            Listen address of Go Ethereum WebSocket API.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 8546;
+          description = ''
+            Port number of Go Ethereum WebSocket API.
+          '';
+        };
+
+        apis = mkOption {
+          type = types.nullOr (types.listOf types.str);
+          default = null;
+          description = ''
+            APIs to enable over WebSocket
+          '';
+          example = ["net" "eth"];
+        };
+      };
+
+      metrics = {
+        enable = lib.mkEnableOption "Go Ethereum prometheus metrics";
+        address = mkOption {
+          type = types.str;
+          default = "127.0.0.1";
+          description = ''
+            Listen address of Go Ethereum metrics service.
+          '';
+        };
+
+        port = mkOption {
+          type = types.port;
+          default = 6060;
+          description = ''
+            Port number of Go Ethereum metrics service.
+          '';
+        };
+      };
+
+      network = mkOption {
+        type = types.enum [ "mainnet" "goerli" "inkeby" "yolov2" "ropsten" ];
+        default = "mainnet";
+        description = ''
+          The network to connect to. Mainnet is the default ethereum network.
+        '';
+      };
+
+      syncmode = mkOption {
+        type = types.enum [ "fast" "full" "light" ];
+        default = "fast";
+        description = ''
+           Blockchain sync mode.
+        '';
+      };
+
+      gcmode = mkOption {
+        type = types.enum [ "full" "archive" ];
+        default = "full";
+        description = ''
+           Blockchain garbage collection mode.
+        '';
+      };
+
+      extraArgs = mkOption {
+        type = types.str;
+        description = ''
+          Additional arguments passed to Go Ethereum.
+        '';
+        default = "";
+      };
+
+      package = mkOption {
+        default = pkgs.go-ethereum;
```

Comment:
the package is called `go-ethereum`. shouldn't we call the service the same?

Reply:
Looking at [other distros](https://repology.org/project/go:ethereum/versions) there seems to be no consesus if the package should be called geth or go-ethereum.

However I think we should name the service `geth` as that is the name you find in almost all guides including a popular [eth2 staking guide](https://someresat.medium.com/guide-to-staking-on-ethereum-2-0-ubuntu-lighthouse-41de20513b12). I think this is what most people expect.

I would rather propose that we rename the package to `geth` to make it more consistent.

Reply:
fair enough. i don't have a strong opinion on which of the two should it be, but i think being the same is preferable.

Reply:
The `go-ethereum` package is now renamed to `geth`.

Pinging maintainers for approval @adisbladis @lionello @xrelkd @RaghavSood

Reply:
Per @adisbladis comment it makes sense to keep both service and package as is. As the geth service is just one think provided by the go-ethereum package.